addons/sale/models/sale.py

```diff
@@ -1367,14 +1376,14 @@ def invoice_line_create(self, invoice_id, qty):
             :param qty: float quantity to invoice
             :returns recordset of account.invoice.line created
         """
-        invoice_lines = self.env['account.invoice.line']
+        vals_list = []
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         for line in self:
             if not float_is_zero(qty, precision_digits=precision) or not line.product_id:
                 vals = line._prepare_invoice_line(qty=qty)
                 vals.update({'invoice_id': invoice_id, 'sale_line_ids': [(6, 0, [line.id])]})
-                invoice_lines |= self.env['account.invoice.line'].create(vals)
-        return invoice_lines
+                vals_list.append(vals)
+        return vals_list
```

Comment:
This is an API change: the function does something different and returns something different.
Please create a secondary method `invoice_line_create_vals` that returns a list of values.
You can then mark `invoice_line_create` as deprecated, so we remove it in the future.

Reply:
True